inference-engine/include/details/ie_alias_map.hpp

```diff
@@ -0,0 +1,274 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header file for the AliasMap class
+ *
+ * @file ie_alias_map.hpp
+ */
+
+#pragma once
+
+#include <map>
+#include <unordered_set>
+
+namespace InferenceEngine {
+namespace details {
+
+template<
+    class Key,
+    class T,
+    class Compare = std::less<Key>,
+    class Allocator = std::allocator<std::pair<const Key, T> >
+> class AliasMap {
```

Comment:
can we inherit from std::map?

Reply:
It would be the best variant, but no, we cannot, `std::map` doesn't have a virtual destructor